src/main/webapp/app/shared/components/code-button/code-button.component.ts

```diff
@@ -120,14 +123,47 @@ export class CodeButtonComponent implements OnInit, OnChanges {
         if (this.useSsh && this.sshEnabled && this.sshTemplateUrl) {
             return this.getSshCloneUrl(this.getRepositoryUri()) || this.getRepositoryUri();
         }
-
         if (this.isTeamParticipation) {
             return this.addCredentialsToHttpUrl(this.repositoryUriForTeam(this.getRepositoryUri()), insertPlaceholder);
         }
 
         return this.addCredentialsToHttpUrl(this.getRepositoryUri(), insertPlaceholder);
     }
 
+    /**
+     * Loads the vcsAccessToken for a participation from the server. If none exists, sens a request to create one
+     */
+    loadVcsAccessToken() {
+        if (this.useVersionControlAccessToken && this.localVCEnabled && this.activeParticipation?.id && this.user) {
+            this.accountService.getVcsAccessToken(this.activeParticipation.id).subscribe({
```

Comment:
As I interpret this code, the `VcsAccessToken` is always fetched from the server here, right? There is no scenario in which the token can be delivered with the initial exercise or participation data, right? ☺️

Reply:
I'm pretty sure it would be possible somehow, but I'm not sure it would be a great idea from a maintainability perspective